File: Java/Desktop/NewLab8/src/by/bntu/fitr/povt/cng/model/entity/container/JSFContainer/JSFLinkedTree.java
package by.bntu.fitr.povt.cng.model.entity.container.JSFContainer;

import by.bntu.fitr.povt.cng.model.entity.abstracts.Toy;

import java.util.Arrays;
import java.util.Iterator;
import java.util.LinkedList;

public class JSFLinkedTree implements JSFTrees{

    private LinkedList<Toy> toys = new LinkedList<>();

    public JSFLinkedTree(Toy...ts) {
        toys.addAll(Arrays.asList(ts));
    }

    public JSFLinkedTree(){

    }

    public Toy[] getMasOfObjects(){
        return (Toy[]) toys.toArray();
    }

    public Toy getToy(int index){
        return toys.get(index);
    }

    public void changeToy(Toy toy, int index){
        toys.remove(index);
        toys.add(index, toy);
    }

    public void remove(int index){
        toys.remove(index);
    }

    public void add(Toy toy){
        toys.add(toy);
    }

    public int getSizeOfTree(){
        return toys.size();
    }

    @Override
    public int hashCode() {
        int result = 0;
        for (Toy toy : toys){
            result += toy.hashCode();
        }
        return result;
    }

    @Override
    public boolean equals(Object obj) {
        if (obj == null && !(obj instanceof JSFArrayTree)){
            return false;
        }
        return obj == this && this.hashCode() == obj.hashCode() && obj.getClass() == this.getClass();
    }

    @Override
    public Iterator<Toy> iterator() {
        return toys.iterator();
    }

    @Override
    public String toString() {
        return toys.toString();
    }

}

File: Csharp/Lab8/Lab8/Lab8/cng/Model/Entity/Container/JSFContainer/JSFLinkedTree.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Lab8.cng.Model.Entity.Abstracts;

namespace Lab8.cng.Model.Entity.Container.JSFContainer
{
    public class JSFLinkedTree : IJsfTree
    {
        public JSFLinkedTree(Toy []toys)
        {
            foreach (var toy in toys)
            {
                Add(toy);
            }
        }
        
        public JSFLinkedTree(){}

        private LinkedList<Toy> _toys = new LinkedList<Toy>();
        
        public IEnumerator<Toy> GetEnumerator()
        {
            return _toys.GetEnumerator();
        }

        IEnumerator IEnumerable.GetEnumerator()
        {
            return _toys.GetEnumerator();
        }

        public int GetSizeOfTree()
        {
            return _toys.Count;
        }

        public void Remove(int index)
        {
            Toy[] objects = _toys.ToArray();
            _toys.Remove(objects[index]);
        }

        public void ChangeToy(Toy toy, int index)
        {
            Toy[] objects = _toys.ToArray();
            objects[index] = toy;
            _toys = new LinkedList<Toy>(objects);
        }

        public void Add(Toy toy)
        {
            _toys.AddLast(toy);
        }

        public Toy[] GetMasOfObjects()
        {
            return _toys.ToArray();
        }

        public Toy GetToy(int index)
        {
            return _toys.ToArray()[index];
        }

        public override string ToString()
        {
            StringBuilder stringBuilder = new StringBuilder();
            foreach (var toy in _toys)
            {
                stringBuilder.Append(toy).Append("\n");
            }

            return stringBuilder.ToString();
        }
    }
}